packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -186,8 +161,14 @@ export default Rule.Atomic.of<Page, Element>({
               );
               const distinguishableContrast = Set.from(
                 Array.from(nonLinkElements)
-                  .map((container) =>
-                    getPairwiseContrast(container, link, device, context)
+                  .map(
+                    (container) => {
+                      const pairings = pairwiseContrastCache.get(
+                        container,
+                        Cache.empty
+                      ).get(link, () => []);
```

Comment:
We should (re)-do the computation here. If, for some reason, the cache is empty when we reach that point, then we do need to compute the contrast and finding it.

Since that contrast computation is duplicated between the two places that use the cache, extracting it to a function will help (and after all the `Cache.get(…)` when calling it we do have `device`, `context`, `link` and `container` already available, so it should be the simple version of it.